Modules/mathmodule.c

```diff
@@ -3914,7 +3916,65 @@ math_nextafter_impl(PyObject *module, double x, double y)
         return PyFloat_FromDouble(y);
     }
 #endif
-    return PyFloat_FromDouble(nextafter(x, y));
+    // fast path:
+    if (steps == 1) {
+        return PyFloat_FromDouble(nextafter(x, y));
+    }
+    if (steps < 0) {
+        PyErr_SetString(PyExc_ValueError, "steps must be >= 0");
+        return NULL;
+    }
+    if (steps == 0)
+        return PyFloat_FromDouble(x);
+    if (Py_IS_NAN(x) || Py_IS_NAN(y))
```

Comment:
Let's follow the IEEE 754 recommendations (§6.2.3, "NaN propagation") to return one of the inputs in this case (so return `x` if `x` is a NaN, and `y` if `y` is a NaN but `x` is not).

Reply:
Sure.  I was going by what https://github.com/bminor/musl/blob/f5f55d6589940fd2c2188d76686efe3a530e64e0/src/math/nextafter.c#L9 does as an inspiration.  But what you says probably makes more sense.